blivet/iscsi.py

```diff
@@ -66,10 +66,28 @@ def has_iscsi():
     return True
 
 
-NodeInfo = namedtuple("NodeInfo", ["name", "tpgt", "address", "port", "iface"])
 TargetInfo = namedtuple("TargetInfo", ["ipaddr", "port"])
 
 
+class NodeInfo(object):
+    """Simple representation of node information."""
+    def __init__(self, name, tpgt, address, port, iface):
+        self.name = name
+        self.tpgt = tpgt
+        self.address = address
+        self.port = port
+        self.iface = iface
+        # These get set by log_into_node: note _login does *NOT*
+        self.username = None
+        self.password = None
+        self.r_username = None
+        self.r_password = None
+
+    def infotuple(self):
```

Comment:
someone (I think kparal) was very rude about my habit of using read-only properties once, so I tend to avoid introducing them to other people's code, though I still use them all over my own =) But if you want it that way, sure, can do.